### PR TITLE
Task/update apiv2

### PR DESCRIPTION
**Pull request by vijayyadav06**
#### Updated Swagger

### CODE REVIEW

File: api/controllers/services/ckan/agencyService.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ class AgencyService extends AbstractService {
         this.agencyResourceId = this.config.agency_resource;
```
Comment (vijayyadav06) [inline api/controllers/services/ckan/agencyService.js]: Leaving all the ckan services as it is. added another folder scosv2 and copied the updated versions of the services there.

File: api/utils/request.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ class RequestUtils {
      * @param req Request
      */
     getPagingParams(req) {
-        const limit = req.query.limit || 100;
+        const limit = req.query.limit || 500;
```
Comment (vijayyadav06) [inline api/utils/request.js]: So I noticed that we were not returning all the locations hits in the API response. The API is designed to receive the `limit` as a parameter in request, although that param is not being passed in the calls coming from UI the api is defaulting to 100 and not returning all locations that were queried.  hence locations like REEB was not showing up. As a workaround increasing it to 500, although @mihiramin89 we need to fix it in frontend.
for example - 
current state (return only 100 results as per default) -
`http://localhost:3000/api/v2/location?taxonomyId=5&agencyId=11110,30525,30525,13800,29900,23560,23550,28500,28500,40500,39650,39100,7800,7700,5120,4124006,5590927,8700,100,5300,11110,45350,30232,24300,50100,41400,26200,7800,40850,48325,50025,50085,7880,4549055,45300,39650,10800,101478,16000,19700,23560,1164300,29900,1610753,48325,46570,2950,10775,19700,36100,29900,29800,1013310,23560,19700,922790,39100,13800,50350,50350,10800,2950,23560,7800,48325,30232,4781935,4200,23560,35325,45350,46575,6392008,5300,2157603,2157603,39650,26200,13800,50025,45350,50025,2157603,6009842,15500,5145,5250,7800,1164300,26200,3974916,4873121,4132490,4132490,4149423,29800,5913514,5913236,5388949,5750620`
vs  below fetches 269 records.
`http://localhost:3000/api/v2/location?taxonomyId=5&agencyId=11110,30525,30525,13800,29900,23560,23550,28500,28500,40500,39650,39100,7800,7700,5120,4124006,5590927,8700,100,5300,11110,45350,30232,24300,50100,41400,26200,7800,40850,48325,50025,50085,7880,4549055,45300,39650,10800,101478,16000,19700,23560,1164300,29900,1610753,48325,46570,2950,10775,19700,36100,29900,29800,1013310,23560,19700,922790,39100,13800,50350,50350,10800,2950,23560,7800,48325,30232,4781935,4200,23560,35325,45350,46575,6392008,5300,2157603,2157603,39650,26200,13800,50025,45350,50025,2157603,6009842,15500,5145,5250,7800,1164300,26200,3974916,4873121,4132490,4132490,4149423,29800,5913514,5913236,5388949,5750620&limit=1000`

File: package.json [hunk 3]
```diff
@@ -51,6 +51,7 @@
     "cors": "^2.8.5",
     "csv": "^5.1.1",
     "express": "^4.16.3",
+    "lodash": "^4.17.14",
```
Comment (vijayyadav06) [inline package.json]: CVE vulnerability reported by github in previous version of lodash.

File: api/controllers/services/scosv2/locationService.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+class LocationService extends AbstractService {
+
+    constructor() {
+        super("agency_location")
+        this.agencyLocationResourceId = this.config.agency_location_resource
+        this.serviceLocationResourceId = this.config.service_location_resource
+        this.uri = `${this.host}/api/v1/organization/handson_central_ohio/dataset/location_consolidated/query?_format=json`
+
+        this.queryFields = `
```
Comment (mihiramin89) [inline api/controllers/services/scosv2/locationService.js]: is there a reason we are just returning specific columns? Should we return it all so the UI or whatever is calling this has access to everything? I am fine either way, just figured i would bring it up.

File: api/controllers/services/scosv2/locationService.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        return mapper(body);
+    }
+}
+
+/**
+ * Remove duplicate array elements
+ * @param arr Array with duplicates
+ * @return unique array
+ */
+function unique(arr) {
```
Comment (mihiramin89) [inline api/controllers/services/scosv2/locationService.js]: could this be accomplished with lodash uniq function? https://lodash.com/docs/#uniq

File: api/controllers/models/taxonomy.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
 class Taxonomy {
```
Comment (mihiramin89) [inline api/controllers/models/taxonomy.js]: very minor, but this file has inconsistent use of semicolons.

File: api/controllers/models/agency.js [hunk 7]
```diff
@@ -1,9 +1,9 @@
 class Agency {
     constructor(res) {
```
Comment (mihiramin89) [inline api/controllers/models/agency.js]: same here. inconsistent use of semicolons.

File: api/controllers/services/scosv2/locationService.js (hunk 4 above)
Comment (vijayyadav06) [reply api/controllers/services/scosv2/locationService.js]: actually this is a dead code, using vscode now that has eslint built in to alert.

File: api/controllers/services/scosv2/locationService.js (hunk 5 above)
Comment (vijayyadav06) [reply api/controllers/services/scosv2/locationService.js]: good point let me try that!

File: api/controllers/models/agency.js (hunk 7 above)
Comment (vijayyadav06) [reply api/controllers/models/agency.js]: removed, should run eslint on all files to remove may be later.

File: api/controllers/models/taxonomy.js (hunk 6 above)
Comment (vijayyadav06) [reply api/controllers/models/taxonomy.js]: done

File: api/controllers/services/scosv2/locationService.js (hunk 5 above)
Comment (vijayyadav06) [reply api/controllers/services/scosv2/locationService.js]: done

**Pull request merged**